### ISSUE TITLE
[Feature] Improving set to be a nested setting function

### ISSUE THREAD

**Issue by s7dhansh** [jagi/meteor-astronomy#527]
What do you think about a `set` function, which sets selectively, i.e. `doc.set({a: {b: {c: 1}}})` behaves like `doc.set("a.b.c": 1)`. So, if doc.a already exists, it does not overwrite it, but modifies it's `b` key. This would be a great step towards using `doc.set(formdata)` as many a times, partial forms exist to modify the document partially. In my apps, I use such a function `setRecursive` which I use without fearing any overwrite.

@jagi Would there be any pitfalls of such a function?

**Comment by lukejagodzinski** [comment]
Hmm can you give me specific example when that would be useful? You said partial forms, but I would like to see an example. I will definitely not change how set method works, because it could break some apps using Astronomy. If such a feature had to be implemented, it would have to be done by an extra option passed to the set method or different method.

**Comment by s7dhansh** [comment]
Yes, true, changing the behavior will cause a havoc. But an extra option would be nice.

Let's say doc has a key `location`:

```
Location = Class.create({
    name: "Location",
    fields: {
        address: String,
        pin: Number,
                landMark: String,
                ....
    }
})
```

Now if I want to give someone the ability to edit everything except the pincode, I can not directly use `doc.set("location", formData)`. I will have to set each field separately.
Or I put pincode in a hidden/disabled field and do extra client checks.

`doc.set("location", formData, {partial: true})` can be a savior. And with a simple `check(formData, ....)` will ensure security as well.

Such cases become common in complex apps, wherein there are mini forms of one collection on a page of another collection.

**Comment by lukejagodzinski** [comment]
Ok I had in my mind a different use case. So actually what you want is the "merge" feature, to make cases like this work properly.

``` js
user.address = {
  city: 'New York',
  state: 'NY'
};

var data = {
  city: 'San Francisco'
};

user.set('address', data, {
  merge: true
});
```

It would work similarly as the lodash's `_.merge()` function which merges properties instead of overriding them. But I assume you would also want this code to work in the same way.

``` js
var data = {
  address: {
    city: 'San Francisco'
  }
};

user.set(data, {
  merge: true
});
```

So in neither cases it would override the `state` property. I like this feature and if it would work for you I would be for making it available as the option for the `set` method.

**Comment by s7dhansh** [comment]
True. Awesome!

**Comment by lukejagodzinski** [comment]
Feature implemented and will appear with the 2.3 release :). I'm closing this for now.
